File: flutter_application_1/lib/services/schedule_service.dart
import '../models/schedule.dart';
import 'api_client.dart';

class ScheduleService {
  static Future<List<Schedule>> getByUserId(int userId) async {
    final json = await ApiClient.get('/schedules/user/$userId');
    return (json as List? ?? [])
        .map((e) => Schedule.fromJson(e as Map<String, dynamic>))
        .toList();
  }

  static Future<Schedule> create(
      Schedule schedule, Map<String, dynamic> userRef) async {
    final json =
        await ApiClient.post('/schedules', body: schedule.toCreateJson(userRef));
    return Schedule.fromJson(json as Map<String, dynamic>);
  }

  static Future<Schedule> cancel(int id) async {
    final json = await ApiClient.put('/schedules/$id/cancel');
    return Schedule.fromJson(json as Map<String, dynamic>);
  }
}

File: ecolink-front/src/app/core/services/schedule.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Schedule, ScheduleStatus } from '../models/index';
import { environment } from '../../../environments/environment';

@Injectable({ providedIn: 'root' })
export class ScheduleService {
  private base = `${environment.apiUrl}/schedules`;
  constructor(private http: HttpClient) {}

  create(schedule: Partial<Schedule>): Observable<Schedule> {
    return this.http.post<Schedule>(this.base, schedule);
  }
  getById(id: number): Observable<Schedule> {
    return this.http.get<Schedule>(`${this.base}/${id}`);
  }
  getByUserId(userId: number): Observable<Schedule[]> {
    return this.http.get<Schedule[]>(`${this.base}/user/${userId}`);
  }
  getByStatus(status: ScheduleStatus): Observable<Schedule[]> {
    return this.http.get<Schedule[]>(`${this.base}/status/${status}`);
  }
  update(id: number, schedule: Partial<Schedule>): Observable<Schedule> {
    return this.http.put<Schedule>(`${this.base}/${id}`, schedule);
  }
  cancel(id: number): Observable<Schedule> {
    return this.http.put<Schedule>(`${this.base}/${id}/cancel`, null);
  }
}
